GitUI/Theming/ThemeRepository.cs

```diff
@@ -86,19 +89,58 @@ private IEnumerable<ThemeId> GetUserCustomizedThemeIds()
                 : Enumerable.Empty<ThemeId>();
         }
 
+        string ICssUrlResolver.ResolveCssUrl(string url)
+        {
+            if (url.EndsWith(Extension, StringComparison.OrdinalIgnoreCase))
+            {
+                url = url.Substring(0, url.Length - Extension.Length);
+            }
+
+            ThemeId id = url.StartsWith(CssVariableUserThemesDirectory)
+                ? new ThemeId(url.Substring(CssVariableUserThemesDirectory.Length), isBuiltin: false)
+                : new ThemeId(url, isBuiltin: true);
+
+            try
+            {
+                return GetPath(id);
+            }
+            catch (ThemeNotFoundException)
+            {
+                return null;
+            }
+        }
+
         private string GetPath(ThemeId id)
         {
+            string path;
             if (id.IsBuiltin)
             {
-                return Path.Combine(AppThemesDirectory, id.Name + Extension);
+                path = Path.Combine(AppThemesDirectory, id.Name + Extension);
             }
+            else
+            {
+                if (UserThemesDirectory == null)
+                {
+                    throw new ThemeNotFoundException("There is no directory for custom user themes in portable mode");
```

Comment:
I think IOE is more appropriate here:
```suggestion
                    throw new InvalidOperationException("Portable mode only supports local themes");
```
Alternatively, may be `FileLoadException`?

Reply:
ok, InvalidOperationException